spec/app/domain/authentication/authn-azure/validate_application_identity_spec.rb

```diff
@@ -0,0 +1,413 @@
+          end
+          subject do
+            Authentication::AuthnAzure::ValidateApplicationIdentity.new(
+              resource_class:             resource_class,
+              validate_azure_annotations: validate_azure_annotations,
+            ).call(
+              account:               account,
+              service_id:            test_service_id,
+              username:              username,
+              xms_mirid_token_field: xms_mirid_token_field_system_assigned_identity,
+              oid_token_field:       oid_token_field
+            )
+          end
+
+          it "raises an error" do
+            expect { subject }.to raise_error(::Errors::Authentication::AuthnAzure::InvalidApplicationIdentity)
+          end
+        end
+
+        context "where the user assigned identity does not match the application identity" do
+          before(:each) do
+            allow(host).to receive(:annotations)
+                             .and_return([subscription_id_annotation, resource_group_annotation, mismatched_user_assigned_identity_annotation])
+          end
+          subject do
+            Authentication::AuthnAzure::ValidateApplicationIdentity.new(
+              resource_class:             resource_class,
+              validate_azure_annotations: validate_azure_annotations,
+            ).call(
+              account:               account,
+              service_id:            test_service_id,
+              username:              username,
+              xms_mirid_token_field: xms_mirid_token_field_system_assigned_identity,
+              oid_token_field:       oid_token_field
+            )
+          end
+
+          it "raises an error" do
+            expect { subject }.to raise_error(::Errors::Authentication::AuthnAzure::InvalidApplicationIdentity)
+          end
+        end
+
+        context "where the system assigned identity does not match the application identity" do
+          before(:each) do
+            allow(host).to receive(:annotations)
+                             .and_return([subscription_id_annotation, resource_group_annotation, mismatched_system_assigned_identity_annotation])
+          end
+          subject do
+            Authentication::AuthnAzure::ValidateApplicationIdentity.new(
+              resource_class:             resource_class,
+              validate_azure_annotations: validate_azure_annotations,
+            ).call(
+              account:               account,
+              service_id:            test_service_id,
+              username:              username,
+              xms_mirid_token_field: xms_mirid_token_field_system_assigned_identity,
+              oid_token_field:       oid_token_field
+            )
+          end
+
+          it "raises an error" do
+            expect { subject }.to raise_error(::Errors::Authentication::AuthnAzure::InvalidApplicationIdentity)
+          end
+        end
+
+        context "where the xms mirid claim does not begin with a /" do
```

Comment:
this is misplaced again. Why is it under `"and an Azure token with non-matching data"`?